analyses/compare-gistic/util/GISTIC-comparison-functions.R

```diff
@@ -0,0 +1,345 @@
+  amp_genes_df <- get_genes_object(amp_genes_file,
+                                   include_peak_info = TRUE)
+  
+  del_genes_df <- get_genes_object(del_genes_file,
+                                   include_peak_info = TRUE)
+  
+  # Bind the rows from the above data.frames into one data.frame
+  final_df <- rbind(amp_genes_df, del_genes_df)
+  
+  
+  # Wrangle the GISTIC all lesions data to be in a comparable format with our CN calls
+  gistic_all_lesions_df <- gistic_all_lesions_df %>%
+    dplyr::select(
+      -c(
+        "Descriptor",
+        "Peak Limits",
+        "Region Limits",
+        "q values",
+        "Residual q values after removing segments shared with higher peaks",
+        "Broad or Focal",
+        "Amplitude Threshold"
+      )
+    ) %>%
+    as.data.frame() %>%
+    tidyr::gather("Kids_First_Biospecimen_ID",
+                  "status",
+                  -`Unique Name`,
+                  -`Wide Peak Limits`) %>%
+    arrange(desc(`Wide Peak Limits`)) %>%
+    dplyr::mutate(status = dplyr::case_when(status < 0 ~ "loss",
+                                            status > 0 ~ "gain",
+                                            status == 0 ~ "neutral")) %>%
+    dplyr::distinct()
+  
+  # Keep just the range information in the `Wide Peak Limits` column
+  # (In order to merge with the amp/del genes data.frame)
+  gistic_all_lesions_df$`Wide Peak Limits` <-
+    gsub("[(].*", "", gistic_all_lesions_df$`Wide Peak Limits`)
+  
+  # Keep just the detection peak name
+  gistic_all_lesions_df$`Unique Name` <-
+    gsub(" -.*", "", gistic_all_lesions_df$`Unique Name`)
+  
+  # Merge the data from the `all_lesions.conf_90.txt` file with the amp/del gene
+  # data.frame prepped above
+  final_df <- final_df %>%
+    dplyr::left_join(gistic_all_lesions_df, by = c("peak_region" = "Wide Peak Limits")) %>%
+    dplyr::select(gene_symbol, Kids_First_Biospecimen_ID, status, detection_peak = `Unique Name`) %>%
+    dplyr::distinct()
+  
+  # Save data.frame to file
+  readr::write_tsv(final_df, file.path(results_dir, output_filename))
+  
+}
+
+prepare_cytoband_level_gistic <- function(all_lesions_file,
+                                          output_filename) {
+  
+  # Given the file path to GISTIC's `all_lesion.conf_90.txt` file,
+  # read in and tidy this data into a data.frame that contains
+  # the cytoband, sample IDs, and their corresponding CN status call.
+  #
+  # Args:
+  #   all_lesions_file: file path to GISTIC's `all_lesion.conf_90.txt` file
+  #   output_filename: string to save the output file as
+  #
+  # Return:
+  #   final_df: data.frame with the relevant data from the `all_lesions` file
+  
+  # Read in files
+  gistic_all_lesions_df <- data.table::fread(all_lesions_file,
+                                             data.table = FALSE)
+  
+  
+  # Wrangle the GISTIC all lesions data to be in a comparable format with our CN calls
+  gistic_all_lesions_df <- gistic_all_lesions_df %>%
```

Comment:
This section looks familiar! :) We could combine `prepare_gene_level_gistic` and `prepare_cytoband_level_gistic` into one function and just add an additional option of either `gene` or `cytoband` but honestly, I think the way you have it is fine too. It's clear this way.